extensions/amp-document-recommendations/0.1/amp-document-recommendations.js

```diff
@@ -229,6 +254,48 @@ export class AmpDocumentRecommendations extends AMP.BaseElement {
     this.appendNextArticle_();
     return Promise.resolve();
   }
+
+  /**
+   * Handles updates from the {@code PositionObserver} indicating a change in
+   * the position of a recommendation unit in the viewport.
+   * @param {number} i Index of the documentRef this recommendation unit is
+   *     attached to.
+   * @param
+   * {!../../../src/service/position-observer/position-observer-worker.PositionInViewportEntryDef}
+   *     position Position of the current recommendation unit in the viewport.
+   */
+  positionUpdate_(i, position) {
+    // We're only interested when the recommendations exit the viewport
```

Comment:
The other PR uses a new custom element and its layout callback to trigger rendering a new recommendation. Can you use the position observer instead to implement that behavior?

Reply:
PositionObserver only fires when the element is in the viewport. We need an events before the element is visible so we can preload.